sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/util/TargetConnectionChecker.java

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.owaspzapwrapper.util;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URI;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mercedesbenz.sechub.owaspzapwrapper.config.ProxyInformation;
+
+/**
+ * This class is used to test if a target URI is reachable. This way we can stop
+ * the scan before it starts in case the target URI cannot be reached.
+ *
+ */
+public class TargetConnectionChecker {
+    private static final Logger LOG = LoggerFactory.getLogger(TargetConnectionChecker.class);
+
+    /**
+     * Tests if site is reachable - no matter if certificate is self signed or not
+     * trusted!
+     *
+     * @param targetUri
+     * @param proxyInformation
+     * @return <code>true</code> when reachable otherwise <code>false</code>
+     */
+    public static boolean isSiteReachable(URI targetUri, ProxyInformation proxyInformation) {
+        URL urlToCheckConnection;
+        try {
+            urlToCheckConnection = targetUri.toURL();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("Target URI: " + targetUri + " could not be transformed into URL!");
+        }
+        if ("https".equals(urlToCheckConnection.getProtocol())) {
+            TrustManager[] trustAllCerts = createTrustAllCertsManager();
+
+            installAllTrustingTrustManager(urlToCheckConnection, trustAllCerts);
+
+            HostnameVerifier allHostsValid = createAllTrustingHostNameVerifier();
+            // Install the all-trusting host verifier
+            HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
```

Comment:
Critical: do NOT install the via the static method! Please look into `TrustAllSupport` of sechub.

Reason: Doing this EVERY standard ssl context URL connection is trusted afterwards - by EVERY url call. So when somebody uses the wrapper as a library after the reachable check, every other url fetch where a url connection is done standard SSL context, every URL would be trusted as well!